packages/@lwc/engine-dom/src/rendererTypecheck.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+// The point of this file is to typecheck that `@lwc/renderer-abstract` has the same API
+// shape as our renderer. It's unused, but just having the file forces TypeScript to typecheck.
+import * as abstractRenderer from '@lwc/renderer-abstract';
+
+import * as renderer from './renderer';
+
+let typecheckedRenderer = renderer;
+typecheckedRenderer = abstractRenderer;
+
+// eslint-disable-next-line no-console
+console.log(typecheckedRenderer); // use the variable so TypeScript doesn't complain
```

Comment:
There is probably a more elegant way to do this typechecking, but I couldn't find it. TypeScript doesn't seem to have a concept of "abstract interface for a whole module, not just an object."

Luckily we can just `import * as foo from "module"` and compare the types on the imported object. This will fail if there is any discrepancy between the abstract renderer and the implemented renderer.